web/webhook.go

```diff
@@ -29,9 +29,16 @@ func incomingWebhook(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	var err *model.AppError
 	incomingWebhookPayload := &model.IncomingWebhookRequest{}
-	mediaType, _, mimeErr := mime.ParseMediaType(r.Header.Get("Content-Type"))
-	if mimeErr != nil && mimeErr != mime.ErrInvalidMediaParameter {
-		c.Err = model.NewAppError("incomingWebhook", "api.webhook.incoming.error", nil, mimeErr.Error(), http.StatusBadRequest)
+	contentType := r.Header.Get("Content-Type")
+	mediaType, _, mimeErr := mime.ParseMediaType(contentType)
+	// Having an empty content-type is fine given that is an optional header
+	if contentType != "" && mimeErr != nil && mimeErr != mime.ErrInvalidMediaParameter {
```

Comment:
Can we clarify the logic here to make the `mediaType` parsing somewhat more explicit on the `contentType` not being empty?
```
var mediaType string
contentType := r.Header.Get("Content-Type")
if contentType != "" {
        mediaType, _, mimeErr := mime.ParseMediaType(contentType)
        if mimeErr != nil && mimeErr != mime.ErrInvalidMediaParameter {
                c.Err = ....
        }
}
```

Reply:
Yes, sure!